client/src/reports/reportgenerator.cpp

```diff
@@ -73,34 +76,30 @@ void ReportGenerator::addHeader(KDReports::Report &report)
     header.addVariable(KDReports::PageCount);
 }
 
-void ReportGenerator::generateListReport(QAbstractItemModel *model, const QString &title,
-                                         const QString &subTitle, QWidget *parent)
+std::unique_ptr<KDReports::Report> ReportGenerator::generateListReport(QAbstractItemModel *model, const QString &title,
+                                         const QString &subTitle)
 {
-    KDReports::Report report;
-    setupReport(report);
-    addTitle(report, title);
-    addSubTitle(report, subTitle);
+    auto report = std::unique_ptr<KDReports::Report>(new KDReports::Report);
 
-    report.addVerticalSpacing(5);
+    setupReport(*report);
+    addTitle(*report, title);
+    addSubTitle(*report, subTitle);
 
-    report.setParagraphMargins(1, 1, 1, 1);
+    report->addVerticalSpacing(5);
+
+    report->setParagraphMargins(1, 1, 1, 1);
     KDReports::AutoTableElement table(model);
     table.setVerticalHeaderVisible(false);
     // make sure country flags do not get huge
     table.setIconSize(QSize(16, 16));
-    report.addElement(table);
+    report->addElement(table);
+
+    finalizeReport(*report);
 
-    finalizeReport(report, parent);
+    return report;
 }
 
-void ReportGenerator::finalizeReport(KDReports::Report &report, QWidget *parent)
+void ReportGenerator::finalizeReport(KDReports::Report &report)
```

Comment:
this method isn't really worth it anymore, for a single line of code, called from a single location...